### PR TITLE
Http request

### PR DESCRIPTION
**Pull request by lucienvb**

### CODE REVIEW

File: src/HTTPRequest.cpp [hunk 1]
```diff
@@ -0,0 +1,125 @@
+#include "../include/HTTPRequest.hpp"
+#include <sstream>
+#include <string>
+
+HTTPRequest::HTTPRequest() {
```
Comment (saladuit) [inline src/HTTPRequest.cpp:5]: Don't forget to initialise the private member variables with the initialiser list.

File: src/HTTPRequest.cpp [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	std::string word;
+	stream_line >> word;
+	if (word == "GET")
+		_methodType = HTTPMethod::GET;
+	else if (word == "POST")
+		_methodType = HTTPMethod::POST;
+	else if (word == "DELETE")
+		_methodType = HTTPMethod::DELETE;
+	else
+		_methodType = HTTPMethod::UNKNOWN;
```
Comment (saladuit) [inline src/HTTPRequest.cpp:22]: I would substitue this with a logger message e.g.:
Logger &logger = Logger::getInstance();
...
logger.log(WARNING, "Unkown method");

File: src/main.cpp [hunk 3]
```diff
@@ -1,20 +1,43 @@
...
+	// PARSING REQUEST
+	test.parse(REQUEST);
+
+	// LOG INFO
+	std::string header = "Host";
+	logger.log(LogLevel::INFO, "Method: %", (int) test.getMethodType());
+	logger.log(LogLevel::INFO, "Path: %", test.getPath());
+	logger.log(LogLevel::INFO, "Version: %", test.getVersion());
+	logger.log(LogLevel::INFO, "Value (with \"%\" as key): %", header, test.getValue(header));
+	logger.log(LogLevel::INFO, "Body: %", test.getBody());
```
Comment (saladuit) [inline src/main.cpp:24]: Perhaps nice to put all of the logging inside of the HTTP rquest class.

**Review by saladuit — CHANGES_REQUESTED** [review]
Looks good, only a few minor changes and all good!

**Review by lucienvb — COMMENTED** [review]
Adjustments has been processed

**Pull request closed without merging**